code/src/main/java/frc/robot/commands/navcommands/ResetGyro.java

```diff
@@ -0,0 +1,42 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands.navcommands;
+
+import com.google.inject.Inject;
+
+import frc.robot.common.*;
+
+/**
+ * Add your docs here.
+ */
+public class ResetGyro extends TraceableCommand {
+    private final IDrivetrainSubsystem _drivetrain;
+
+    @Inject
+    public ResetGyro(final ILogger logger, IDrivetrainSubsystem drivetrain) {
+        super(logger);
+        _drivetrain = drivetrain;
+
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+    }
+
+    @Override
+    public void execute() {
+        super.execute();
+        _drivetrain.resetGyro();
```

Comment:
Shoudn't this be in the initialize method  so you only reset the gyro once

Reply:
yes that's a bug

Reply:
sorry, i misread that.  no, it's not a bug.  if you look at the isFinished case, it returns true, so it's an instant method.  meaning that the execute method also only runs once